v2/message.go

```diff
@@ -310,6 +310,10 @@ type MessageOptions struct {
 	// this is set, any Partitioner will not be used. This is a pointer to
 	// allow distinguishing between unset and 0.
 	Partition *int32
+
+	// ExpectedOffset set the value of the expected offset after publishing the message.
+	// This is required in case Optimistic Concurrency Control is activted
+	ExpectedOffset int64
```

Comment:
This should default to `-1` wherever `MessageOptions` is created so that the default behavior is to take the next offset.

Reply:
We should also document the behavior of `-1`.

Reply:
For the documentation, I added in in README.md of the client. Not sure where is the better place for this behavior ?